drivers/ethernet/eth_stellaris.c

```diff
@@ -104,27 +104,10 @@ static int eth_stellaris_send(struct device *dev, struct net_pkt *pkt)
 
 	if (dev_data->tx_err) {
 		dev_data->tx_err = false;
-		net_pkt_unref(pkt);
```

Comment:
There is one another net_pkt_unref() at the top that needs to be removed.

Reply:
indeed. Actually that check there is superfluous: a pkt has always buffer when it reaches the driver. since such verification is done in net_core.c:net_send_data() function. I'll update.